src/bun.js/bindings/ErrorStackTrace.cpp

```diff
@@ -0,0 +1,370 @@
+        if (error) {
+            jscStackTrace = error->stackTrace();
+        }
+    }
+
+    if (!jscStackTrace) {
+        return JSCStackTrace();
+    }
+
+    return fromExisting(vm, *jscStackTrace);
+}
+
+JSCStackFrame::JSCStackFrame(JSC::VM& vm, JSC::StackVisitor& visitor)
+    : m_vm(vm)
+    , m_codeBlock(nullptr)
+    , m_bytecodeIndex(JSC::BytecodeIndex())
+    , m_sourceURL(nullptr)
+    , m_functionName(nullptr)
+    , m_isWasmFrame(false)
+    , m_sourcePositionsState(SourcePositionsState::NotCalculated)
+{
+    m_callee = visitor->callee().asCell();
+    m_callFrame = visitor->callFrame();
+
+    // Based on JSC's GetStackTraceFunctor (Interpreter.cpp)
+    if (visitor->isWasmFrame()) {
+        m_wasmFunctionIndexOrName = visitor->wasmFunctionIndexOrName();
+        m_isWasmFrame = true;
+    } else if (!!visitor->codeBlock() && !visitor->codeBlock()->unlinkedCodeBlock()->isBuiltinFunction()) {
+        m_codeBlock = visitor->codeBlock();
+        m_bytecodeIndex = visitor->bytecodeIndex();
+    }
+}
+
+JSCStackFrame::JSCStackFrame(JSC::VM& vm, const JSC::StackFrame& frame)
+    : m_vm(vm)
+    , m_callFrame(nullptr)
+    , m_codeBlock(nullptr)
+    , m_bytecodeIndex(JSC::BytecodeIndex())
+    , m_sourceURL(nullptr)
+    , m_functionName(nullptr)
+    , m_isWasmFrame(false)
+    , m_sourcePositionsState(SourcePositionsState::NotCalculated)
+{
+    m_callee = frame.callee();
+
+    // Based on JSC's GetStackTraceFunctor (Interpreter.cpp)
+    if (frame.isWasmFrame()) {
+        m_wasmFunctionIndexOrName = frame.wasmFunctionIndexOrName();
+        m_isWasmFrame = true;
+    } else {
+        m_codeBlock = frame.codeBlock();
+        if (frame.hasBytecodeIndex()) {
+            m_bytecodeIndex = frame.bytecodeIndex();
+        }
+    }
+}
+
+intptr_t JSCStackFrame::sourceID() const
+{
+    return m_codeBlock ? m_codeBlock->ownerExecutable()->sourceID() : JSC::noSourceID;
+}
+
+JSC::JSString* JSCStackFrame::sourceURL()
+{
+    if (!m_sourceURL) {
+        m_sourceURL = retrieveSourceURL();
+    }
+
+    return m_sourceURL;
+}
+
+JSC::JSString* JSCStackFrame::functionName()
+{
+    if (!m_functionName) {
+        m_functionName = retrieveFunctionName();
+    }
+
+    return m_functionName;
+}
+
+JSC::JSString* JSCStackFrame::typeName()
+{
+    if (!m_typeName) {
+        m_typeName = retrieveTypeName();
+    }
+
+    return m_typeName;
+}
+
+JSCStackFrame::SourcePositions* JSCStackFrame::getSourcePositions()
+{
+    if (SourcePositionsState::NotCalculated == m_sourcePositionsState) {
+        m_sourcePositionsState = calculateSourcePositions() ? SourcePositionsState::Calculated : SourcePositionsState::Failed;
+    }
+
+    return (SourcePositionsState::Calculated == m_sourcePositionsState) ? &m_sourcePositions : nullptr;
+}
+
+ALWAYS_INLINE JSC::JSString* JSCStackFrame::retrieveSourceURL()
+{
+    if (m_isWasmFrame) {
+        return jsString(m_vm, makeString("[wasm code]"_s));
```

Comment:
Let's avoid these dynamic memory allocations

- Grep for `MAKE_STATIC_STRING_IMPL`
- Copy that pattern
- Use `jsOwnedString` instead of `jsString`